### PR TITLE
tenant id proliferation

### PR DESCRIPTION
**Pull request by boriwo**
issues 127 and 128

### CODE REVIEW

File: internal/pkg/plugin/manager.go [hunk 1]
```diff
@@ -577,38 +580,30 @@ func (m *manager) SendersStatus() map[string]SenderStatus {
...
 	return nil
 
 }
 
 // === Helper Functions ==============================================
 
-func (m *manager) mapkey(name string, hash string) string {
-	return name + "/" + hash
+func (m *manager) mapkey(tid tenant.Id, name string, hash string) string {
+	return tid.OrgId + "/" + tid.AppId + "/" + name + "/" + hash
```
Comment (holidaymike) [inline internal/pkg/plugin/manager.go:608]: Why are we not using tid.Key() function? If we agree that we want to allow all characters in orgId and appId, we should stick to a more generic key generation.

File: pkg/receiver/types.go [hunk 2]
```diff
@@ -60,4 +61,5 @@ type Receiver interface {
 	Config() interface{}
 	Name() string
 	Plugin() string
+	Tenant() tenant.Id
```
[the remark this answers is not in the document]
Comment (boriwo) [reply pkg/receiver/types.go:64]: ok - will think about that. you could argue the same for Name() and Plugin() and even Config() - malicious plugin could always do bad stuff.

[the remark this answers is not in the document]
Comment (holidaymike) [reply pkg/receiver/types.go:64]: Yeah, I was thinking about that too. If it is too much work to fix, we should make it an issue to revisit later.

File: internal/pkg/plugin/manager.go (hunk 1 above)
Comment (holidaymike) [reply internal/pkg/plugin/manager.go:608]: Got it. Didn't realize that this just need to be one way. I take back my comment. This is ok

Comment (holidaymike) [reply internal/pkg/plugin/manager.go:608]: I just realize that {"orgId": "foo/foo", "appId":"bar"} will result in the same mapkey as {"orgId":"foo", appId:"foo/bar"}. Granted, this is an edge case, but we can no longer claim that all characters are allowed

File: pkg/receiver/types.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (boriwo) [reply pkg/receiver/types.go:64]: ok

**Pull request merged**